### ISSUE TITLE
Colour estimate

### ISSUE THREAD

**Issue by jamarav** [vincentarelbundock/modelsummary#440]
Hi everybody,

I would like to make some comments about automatic coloring the estimate values. Currently I have found a not very elegant solution for the problem below:

Case: I have a list of models with the same equation but different samples in the adjustment. The first element of the list is the reference model adjusted with all the experimental data available. The rest of models are adjusted with a minumum sample defined with different Design of Experiments methodologies (DoE).

Objective: The objective is compare the coefficients for all the models adjusted with the DoE samples with the reference model. Only if there is a change in the sign of the coefficients.

My solution:

I used the function `tidy_custom.lm` in order to define an extra colum with the correct string formatted to obain the desired color in Latex+KableExtra. Additionally, I incude a proper string in the `estimate` parameter with the glue package. The specific column added for the colour in `tidy_custom` is  `colour.estimate`

This is the source code:

```
tidy_custom.lm <- function(x, conf_level = 0.95,...) {
  #browser()
  s <- summary(x)$coefficients
  
  alpha<-1-conf_level
  
  new_col<-qt(1-alpha/2, df = df.residual(x)) * s[, 'Std. Error'] #Manual calculation confidence interval
  #new_col<-summary(x)$coefficients[, 1] - confint(x, level  = conf_level)[,1] #Confidence interval extracted from confint() function
  
  Ref_coeff<-get("Ref_coeff", envir=globalenv())
  
  tryCatch(
    expr = {
      out <- data.frame(
        term = row.names(s),
        `conf.int.custom` = sigDig(new_col, digits_ = 3),
        `colour.estimate`=ifelse(sign(Ref_coeff) == sign(s[,1]), "{black}", "{red}"))
      return(out)
    },
    error = function(e){
      out <- data.frame(
        term = row.names(s),
        `conf.int.custom` = sigDig(new_col, digits_ = 3))
      return(out)
      
    }
  )
  
  
  
}

#Generate the table
l_models #List of models

Ref_coeff<-l_models[[1]]$coefficients

wraper1<-"{"
wraper2<-"}"
p<-modelsummary(models, gof_map = gm, add_rows = MRE_RMSE_modelsummary(l_models = models, units_RMSE="W", mult=1000, fmt_MRE = 2, fmt_RMSE = 2), estimate = "\\textcolor{colour.estimate}{wraper1}{estimate}{wraper2} \\scriptsize{stars}", fmt = '%.2e', statistic=NULL, stars = stars_modelsummary, escape=F, coef_rename = mycoef_rename, caption = "Title") %>%
kable_styling(latex_options = c("hold_position", "striped", "scale_down")) %>%
    footnote(escape = F, threeparttable = T,
			     alphabet = c("{\\\\footnotesize + p $<$ 0.1, * p $<$ 0.05, ** p $<$ 0.01, *** p $<$ 0.001; Confidence interval of 95\\\\% for regression coefficients;}",
			         "{\\\\footnotesize Temperatures (\\\\degree C);}",
			         "{\\\\footnotesize Energy consumption (kW);}"
			         )
			     )
```

Sorry for not providing a complete example. I just want to comment my solution and explore if there are more efficient alternatives to do this.

Some dissadvantages with this implementation:
-We need to save a vector (`Ref_coeff`) with the reference coefficients in the global environtment. Then invoke this vector in the local environtment of the `tidy.custom` function.
-It depends on assign previously the Ref_coeff` in the global environtment. We may forget to assign it or assign it wrong.
-Some problems with the glue plackage. In latex mainly with "{" and "}". I tested writing directly "\\{" in the estimate parameter but it doesn't works. I need to define previously the variables wraper1 and wraper2. Probably not necessary and the problem is that I don't know how to correctly format the string directly.

Any suggestion for improvement is welcome.

Thanks in advance

**Comment by vincentarelbundock** [comment]
Personally, I would insert the whole `\textcolor` part immediately inside the `tidy_custom` method instead of using a glue string.

Beyond that, I'm not sure exactly what you are asking for.

Is there a bug to report or a specific feature you'd like to request?

**Comment by jamarav** [comment]
Thanks for the quick reply @vincentarelbundock. There is not a bug. It's only comment the best way to solve this and if it's possible to use other approach. Personally I don't like to invoke variables from the global environtment, we need to define previously and the modelsummary table not only depends from the list provided. It could be better if it's possible to avoid this. I'm not sure, I would like only to know if there is an optional approach.

Thanks in advance

**Comment by jamarav** [comment]
Regarding to include directly the `\textcolor` I'm going to include your suggestion, thanks. Any advice about the variables `wraper1` and  `wraper2`? In Latex is necessary to wrap the `estimate` value like `\texcolor{red}{1.5646}`

**Comment by vincentarelbundock** [comment]
I'm not sure why you don't define those inside `tidy_custom`. They wouldn't be in the global environment...

**Comment by jamarav** [comment]
I defined here in order to reuse the function. I need to compile and generate a lot of table and in each table the reference model change (I provide a different list of mdels in each iteration). For that, I need to change the Ref_coeff for each list of models. Another option is to send the Ref_coeff to `tidy_custom` but I'm not sure to include this when I generate the table with  `modelsummay()`

Thanks in advance

**Comment by vincentarelbundock** [comment]
Yeah, frankly I don't think I understand what you want to achieve.

Maybe if you supply a minimal working example with one of the built-in datasets like `mtcars`, I can help, but right now I can't wrap my head around what you want/need.

**Comment by jamarav** [comment]
Sorry for not provide first a simple example. I attached a simple `*.Rnw` file.

Thanks in advance @vincentarelbundock
[Example.zip](https://github.com/vincentarelbundock/modelsummary/files/8125149/Example.zip)

**Comment by vincentarelbundock** [comment]
OK, but the problem is conceptual. I don't understand what you are doing, and I thought it would help to get a SIMPLE reproducible example that illustrates the very MINIMAL functionality you want.

**Comment by jamarav** [comment]
Amazing. I didin't know this feature. I always use modelsummary with a model or list of models. Of course this approach is better and simple. That's why the question, I intuited that there had to be a simple way to do this.


Thank you very much for all your help @vincentarelbundock

**Comment by vincentarelbundock** [comment]
Wonderful!

**Comment by jamarav** [comment]
Only to report your suggestion this is the final solution:

```
sigDig = function(x, digits_ = 3){
    
    x_<-formatC(x, digits = digits_, format = "e")
    
    x_<-paste0(gsub(x = x_, pattern = "e",replacement = "\\\\text{e"), "}")
    
    x_
    
}

check_coeff_with_reference = function(x, digits = 3){
  ref_coeff<-x[[1]]$tidy$estimate
  
  for(i in 1:length(x)){
    coeff<-x[[i]]$tidy$estimate
    
    x[[i]]$tidy$`estimate.custom`<-ifelse(sign(ref_coeff) == sign(coeff), paste0(sigDig(coeff, digits_ = digits)), paste0("\\textcolor{red}{", sigDig(coeff, digits_ = digits),"}"))
    
  }
  return(x)
}

#Generate table

model #list of models

p<-modelsummary(models, output = "modelsummary_list")

p<-check_coeff_with_reference(p)

p<-modelsummary(p, estimate = "{estimate.custom} \\scriptsize{stars}")
 
```

Thanks you very much @vincentarelbundock
